CONTRIBUTING.md

```diff
@@ -0,0 +1,619 @@
+
+```bash
+sed -i '' 's/old/new/g' file
+```
+
+### Building Container Images
+
+Build the main analyzer container:
+
+```bash
+podman build -t quay.io/konveyor/analyzer-lsp:latest -f Dockerfile .
+```
+
+Build external provider containers:
+
+```bash
+# Builds all external providers
+make build-external
+```
+
+## Testing Your Changes
+
+### Running Tests Locally
+
+Run the Go test suite:
+
+```bash
+go test ./...
+```
+
+### Testing with Example Projects
+
+The `examples/` directory contains test projects for each language:
+
+- `examples/java/` - Java test projects
+- `examples/golang/` - Go test projects
+- `examples/nodejs/` - Node.js/TypeScript test projects
+- `examples/python/` - Python test projects
+- `examples/yaml/` - YAML/Kubernetes manifests
+
+### Running Analysis Locally
+
+Create a provider settings file (e.g., `provider_settings.json`):
+
+```json
+{
+  "name": "nodejs",
+  "binaryPath": "/usr/local/bin/typescript-language-server",
+  "address": "127.0.0.1:0",
+  "initConfig": [
+    {
+      "location": "/path/to/your/project",
+      "providerSpecificConfig": {
+        "includedPaths": ["src/"],
+        "excludedPaths": ["node_modules/", "dist/"]
+      }
+    }
+  ]
+}
+```
+
+Run analysis:
+
+```bash
+./kantra \
+  --provider-settings=provider_settings.json \
+  --rules=rule-example.yaml \
+  --output-file=output.yaml \
+  --verbose=1
+```
+
+## Container-Based Development
+
+Container-based testing is the **recommended approach** for comprehensive testing with all providers.
+
+### Why Use Containers?
+
+1. **Consistent Environment** - Same environment as CI/CD
+2. **All Providers Together** - Test interactions between providers
+3. **Resource Isolation** - Prevents provider memory/CPU conflicts
+4. **Reproducible** - Matches production deployment
+
+### Container Testing Workflow
+
+This is the workflow used for regenerating `demo-output.yaml`:
+
+```bash
+# 1. Build external providers with your changes
+make build-external
+
+# 2. Build analyzer-lsp container image
+podman build -t quay.io/konveyor/analyzer-lsp:latest -f Dockerfile .
+
+# 3. Run external providers pod
+make run-external-providers-pod
+
+# 4. Build demo container image
+podman build -f demo-local.Dockerfile -t localhost/testing:latest .
+
+# 5. Run demo image to generate output
+make run-demo-image
+```
+
+### Provider Pod Architecture
+
+The `run-external-providers-pod` target creates a pod named `analyzer` with 6 containers:
+
+- `java-provider` - Port 14650 (Eclipse JDTLS)
+- `generic-provider` - Port 14651 (Go/Python/Node.js)
+- `dotnet-provider` - Port 14652 (.NET)
+- `yq-provider` - Port 14653 (YAML)
+- `golang-dep-provider` - Port 14654 (Go dependencies)
+- `java-dep-provider` - Port 14655 (Java dependencies)
+
+All containers share the `test-data` volume for accessing example projects.
```

Comment:
Can we add a note that the test-data volume is populated by copying from the examples that are built into the analyzer-lsp image?